File: app/src/main/java/com/gystry/pjetpack/model/Feed.java
package com.gystry.pjetpack.model;

import android.text.TextUtils;

import androidx.annotation.Nullable;

import java.io.Serializable;

/**
 * @author gystry
 * 创建日期：2020/7/8 10
 * 邮箱：gystry@163.com
 * 描述：
 */
public class Feed implements Serializable {
    public static final int VIDEO_TYPE=2;//图文
    public static final int IMAGE_TYPE=1;//视频
    public int id;
    public long itemId;
    public int itemType;
    public long createTime;
    public double duration;
    public String feeds_text;
    public long authorId;
    public String activityIcon;
    public String activityText;
    public int width;
    public int height;
    public String url;
    public String cover;

    public User author;
    public Comment topComment;
    public Ugc ugc;

    @Override
    public boolean equals(@Nullable Object obj) {
        if (obj == null || !(obj instanceof Feed))
            return false;
        Feed newFeed = (Feed) obj;
        return id == newFeed.id
                && itemId == newFeed.itemId
                && itemType == newFeed.itemType
                && createTime == newFeed.createTime
                && duration == newFeed.duration
                && TextUtils.equals(feeds_text, newFeed.feeds_text)
                && authorId == newFeed.authorId
                && TextUtils.equals(activityIcon, newFeed.activityIcon)
                && TextUtils.equals(activityText, newFeed.activityText)
                && width == newFeed.width
                && height == newFeed.height
                && TextUtils.equals(url, newFeed.url)
                && TextUtils.equals(cover, newFeed.cover)
                && (author != null && author.equals(newFeed.author))
                && (topComment != null && topComment.equals(newFeed.topComment))
                && (ugc != null && ugc.equals(newFeed.ugc));
    }

}

File: appkt/src/main/java/com/gystry/appkt/model/Feed.kt
package com.gystry.appkt.model

import android.text.TextUtils
import java.io.Serializable

/**
 * @author gystry
 * 创建日期：2021/3/30 11
 * 邮箱：gystry@163.com
 * 描述：
 */
data class Feed(var id: Int = 0,
                var itemId: Long = 0,
                var itemType: Int = 0,
                var createTime: Long = 0,
                var duration: Double = 0.0,
                var feeds_text: String,
                var authorId: Long = 0,
                var activityIcon: String,
                var activityText: String,
                var width: Int = 0,
                var height: Int = 0,
                var url: String,
                var cover: String,
                var author: User,
                var topComment: Comment,
                var ugc: Ugc):Serializable {
    companion object{
        const val TYPE_IMAGE=1
        const val TYPE_VIDEO=2
    }
    override fun equals(@androidx.annotation.Nullable other : Any?): Boolean {
        if (other == null || other !is Feed) return false
        val newFeed:Feed = other
        return (id == newFeed.id && itemId == newFeed.itemId && itemType == newFeed.itemType && createTime == newFeed.createTime && duration == newFeed.duration
                && TextUtils.equals(feeds_text, newFeed.feeds_text)
                && authorId == newFeed.authorId && TextUtils.equals(activityIcon, newFeed.activityIcon)
                && TextUtils.equals(activityText, newFeed.activityText)
                && width ==newFeed. width && height == newFeed.height && TextUtils.equals(url, newFeed.url)
                && TextUtils.equals(cover, newFeed.cover)
                && author != null && newFeed.author != null && author == newFeed.author
                && topComment != null && topComment == newFeed.topComment
                && ugc != null && ugc == newFeed.ugc)
    }
}